### DISCUSSION TITLE
GLOBAL_SCRIPTS Attribute Assignment

### DISCUSSION THREAD

**Discussion by kovitikus** [evennia/evennia#2365] [Evennia Questions] [answered]
So, I made this global script.
```py
'instance_ledger': {
        'typeclass': 'typeclasses.scripts.Script',
        'desc': 'Holds record of all current instances.',
        'persistent': True
    }
```

I reloaded the server and it showed up just fine.

Later on I decided I want to add an attribute available on creation, but I didn't want to create a typeclass just for this script.

So I added a new line to declare my `instances` attribute as an empty dictionary..
```py
    'instance_ledger': {
        'typeclass': 'typeclasses.scripts.Script',
        'desc': 'Holds record of all current instances.',
        'persistent': True,
        'attributes': [('instances', {})]
    }
```
I saved my code.

I killed the previous running script and double checked with the in game command `scripts` that it was indeed gone.
I then used `evennia stop` and `evennia start` to do a cold restart of my server. The script was recreated, but the attribute is nowhere to be found. I checked the admin page and found no attributes.

![script admin page](https://i.imgur.com/muw0dHJ.png)

***

I also attempted to use the in game interactive `py` console to get the script and check on the attribute. It still returned `None`.
```py
my_script = GLOBAL_SCRIPTS.instance_ledger

print(my_script)
```
> <Script instance_ledger>

***

```py
my_attribute = my_script.attributes.get('instances')

print(my_attribute)
```
> None

***

What gives? Why won't this attribute appear for me?

I followed the guidelines for the attributes argument passed into [`create_script()`](https://github.com/evennia/evennia/blob/master/evennia/utils/create.py#L186).
> attributes (list): List of tuples `(key, value)`, `(key, value, category)`,
           `(key, value, category, lockstring)` or
           `(key, value, category, lockstring, default_access)`.

**Comment by strikaco** [comment, marked as answer]
https://github.com/evennia/evennia/blob/f1b38866081641a1b37939ab706df75eceb3c12a/evennia/utils/containers.py#L141-L150

The global script handler appears to ignore attributes altogether.

**Reply by kovitikus** [reply]
Huh, interesting. Thanks for the heads-up.

I know that @Griatch is refactoring them and I'm curious to know if this will change.

There's quite a few arguments missing.

**Reply by Griatch** [reply]
We could certainly add support for all the script-creation kwargs.
